docs/source/authoring_flows.rst

```diff
@@ -213,30 +219,48 @@ The syntax of an expression paramter takes the following form:
 .. code-block:: JSON
 
     {
-      "computed_param.=": "`$.JsonPathExpr1` <op> `$.JsonPathExpr2` <op> ..."
+      "computed_param.=": "<state_val1> <op> <state_val2> <op> ..."
     }
 
 
-The important parts of this expression are the references to the Flow state via
-`JsonPath <https://goessner.net/articles/JsonPath/>`_ expressions, and the
-operations and expression syntax that may be used. Values from the state are
-specified via a JsonPath expression which is surrounded by single "back-quote"
-characters (\`). The full selection capability of JsonPath is supported, so
-entire list values, list indexing, list slicing and so on may be specified in
-the JsonPath.
-
-Values in the expression may also be constant values. It is important to
-remember that within an expression, a string type value must be enclosed in
-quotes. Thus, the expression ``foo + bar`` will be an error as the unquoted
-values ``foo`` and ``bar`` don't represent either a constant or a JsonPath
-value, where as the expression ``"foo" + "bar"`` will result in the expected(?)
-output ``foobar``.
 
 The syntax for the expression largely follows what is expected in common
 expression languages. This includes common arithmetic operators on numeric
 values as well as operations on strings (e.g. string concatenation via a `+`
 operation) and on lists (similarly the `+` operator will concatenate lists).
 
+The values in the state of the flow may be used in the expression and are denoted as ``<state_valN>`` above. For the following description, assume that the input to (or current state of) a Flow Run is as follows:
+
+.. code-block:: JSON
+
+    {
+      "foo": "bar",
+      "list_val": [1, 2, 3],
+      "object_val": {
+        "sub_val1": "embedded",
+        "sub_val2": "also_embedded"
+      }
+    }
+
+The ``state_val`` values can be specified in one of two ways with the first syntax being preferred for long term support in the service.
+
+1. Simple names of the properties in the state of the running flow allowing for indexing into lists and into embedded objects similar to Python. Thus, the following would be a valid expression: ``foo + ' ' + object_val.sub_val1`` which would yield the string ``bar embedded``. Note the use of ``+`` to mean string concatenation and the dot-separated naming of the field of the object.
+
+2. The `JsonPath <https://goessner.net/articles/JsonPath/>`_ syntax
```

Comment:
I wonder if we should still advertise this as an option. I guess it's good for folks who are already working with Flows that use the JSONPath syntax.